controllers/selfnoderemediation_controller.go

```diff
@@ -142,7 +152,22 @@ func (r *SelfNodeRemediationReconciler) Reconcile(ctx context.Context, req ctrl.
 
 	if r.isStoppedByNHC(snr) {
 		r.logger.Info("SNR remediation was stopped by Node Healthcheck")
-		return ctrl.Result{}, nil
+		org := snr.DeepCopy()
+		if err := r.updateSnrProcessingCondition(remediationTerminatedByNHC, snr); err != nil {
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, r.patchSnrStatus(snr, org)
+	}
+
+	//In case snr is being deleted there is no remediation in process
+	if !r.isFencingCompleted(snr) {
```

Comment:
I think this is the wrong place for setting the condition to true. There are some checks following, which will prevent an actual remediation. IMHO the condition should be set to true when at the place where we set the finalizer, and in all other cases to false. That would ensure that NHC tries the next remediation quickly. WDYT?

Reply:
The way I understand it setting the flag to true is merely an indication that SNR has started remediating, it is by no means an assurance that the remediation will be successful.
It is true that setting the flag on a later stage might save time in NHC escalation use cases, but I think it would not reflect the current situation correctly (hiding information both from the user and from NHC).

I am not sure but  by setting the flag later don't  we run into the opposite risk of NHC assuming SNR is prematurely ?

Reply:
as discussed on chat, we might want to revisit this in a follow up

Reply:
Adding a [link](https://redhat-internal.slack.com/archives/C03M5GKJNBA/p1679306209179739) to the slack chat for future me